app/services/foreman_ansible/override_resolver.rb

```diff
@@ -0,0 +1,68 @@
+module ForemanAnsible
+  # Service which resolves override values for hosts and hostgroups
+  class OverrideResolver
+    attr_reader :overrides
+
+    def initialize(resource_id, resource_name, parent_id, variables)
+      @overrides = calculate_variable_overrides(resource_id, resource_name, parent_id, variables)
+    end
+
+    def resolve(ansible_variable)
+      override = @overrides[ansible_variable.id]
+      return unless override
+      override[ansible_variable.key]
+    end
+
+    private
+
+    def calculate_variable_overrides(host_or_hg_id, resource_name, parent_id, ansible_variables)
+      return {} if (!host_or_hg_id && !parent_id) || !resource_name
+      resource = find_resource resource_name, host_or_hg_id, parent_id
+
+      return {} unless resource
+      resolve_overrides resource, ansible_variables
+    end
+
+    def find_resource(resource_name, host_or_hg_id, parent_id)
+      if host_or_hg_id
+        resource_name.constantize.find_by :id => host_or_hg_id
+      else
+        Hostgroup.find_by :id => parent_id
+      end
+    end
+
+    def resolve_overrides(resource, ansible_variables)
+      if resource.is_a? Host::Base
+        ansible_variables.values_hash(resource).raw
+      else
+        hostgroup_values_hash ansible_variables, resource
+      end
+    end
+
+    def hostgroup_values_hash(ansible_variables, hostgroup)
+      ansible_variables.each_with_object({}) do |ansvar, memo|
+        next memo unless ansvar.override
+        hash_result = inherited_hostgroup_value(hostgroup, ansvar)
+        next memo unless hash_result
+        memo[ansvar.id] = {
+          ansvar.key => {
+            :value => hash_result[:lookup_value].value,
+            :element => 'hostgroup',
+            :element_name => hash_result[:hostgroup].name
+          }
+        }
+        memo
+      end
+    end
+
+    def inherited_hostgroup_value(hostgroup, ansvar)
+      return unless ansvar.path_elements.flatten.include?('hostgroup') || Setting['matchers_inheritance']
+
+      hostgroup.path.reverse_each do |hg|
```

Comment:
It is possible I missed an edge case and I am not 100% sure since it has been a while since I wrote this but I believe the intent here is to get the value for hostgroup regardless of whether it is inherited or not. That does not mean we cannot have the 'preview' in UI, because we keep track of the hostgroup name this was resolved for, so it should not be difficult to decide if the value is inherited or not.

Reply:
I mean: you have overriden value on hostgroup, in puppet parameters you see in tooltip what is the value of parent. And you know right away what value to show once you disable the override on the hostgroup. with this, we can't show the tooltip and we need API request in case of override get's disabled. IMHO it can be usefull to have both values inherited and the direct override.

Reply:
I agree that we should not do additional requests if they can be avoided, but right now we do not do any. I think this discussion can be had on the follow-up PR if we face that problem.

What are the changes you would like me to do here?